src/enigma2/data/RecordingEntry.h

```diff
@@ -58,27 +58,42 @@ namespace enigma2
       const std::string& GetChannelName() const { return m_channelName; }
       void SetChannelName(const std::string& value ) { m_channelName = value; }
 
+      int GetChannelUniqueId() const { return m_channelUniqueId; }
+      void SetChannelUniqueId(int value) { m_channelUniqueId = value; }
+
       const std::string& GetDirectory() const { return m_directory; }
       void SetDirectory(const std::string& value ) { m_directory = value; }
 
       const std::string& GetIconPath() const { return m_iconPath; }
       void SetIconPath(const std::string& value ) { m_iconPath = value; }
 
+      bool IsRadio() const { return m_radio; }
+      void SetRadio(bool value) { m_radio = value; }
+
       bool UpdateFrom(TiXmlElement* recordingNode, const std::string &directory, enigma2::Channels &channels);
       void UpdateTo(PVR_RECORDING &left, Channels &channels, bool isInRecordingFolder);
 
     private:
       long TimeStringToSeconds(const std::string &timeString);
 
+      std::shared_ptr<Channel> FindChannel(enigma2::Channels &channels);
+      std::shared_ptr<Channel> GetChannelFromChannelReferenceTag(enigma2::Channels &channels);
+      std::shared_ptr<Channel> GetChannelFromChannelNameSearch(enigma2::Channels &channels);
+      std::shared_ptr<Channel> GetChannelFromChannelNameFuzzySearch(enigma2::Channels &channels);
```

Comment:
make all 4 new methods const?

Reply:
... and make `m_anyChannelTimerSource`, `m_radio` and `m_haveChannelType` `mutable `  => logical constness of `FindChannel`

Reply:
Is mutable a requirement of making the methods const?

Reply:
As FindChannel writes to these members, yes. But there is nothing bad with this. That's what mutable is for - achieve logical constness of a method.

Reply:
Very cool, probably have a few more places I could use this!

Reply:
Done